kratos/processes/calculate_distance_to_path_process.h

```diff
@@ -0,0 +1,404 @@
+{
+    // Defining clearer options
+    constexpr static bool SaveAsHistoricalVariable = true;
+    constexpr static bool SaveAsNonHistoricalVariable = false;
+};
+
+/** 
+ * @class CalculateDistanceToPathProcess
+ * @ingroup KratosCore 
+ * @brief Computes DISTANCE from a path model part to a given model part
+ * @author Vicente Mataix Ferrandiz
+ * @tparam THistorical If the distance is computed as historical or non historical variable
+ */
+template<bool THistorical = true>
+class KRATOS_API(KRATOS_CORE) CalculateDistanceToPathProcess
+    : public Process
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Index type definition
+    typedef std::size_t IndexType;
+    
+    /// Size type definition
+    typedef std::size_t SizeType;
+
+    /// The definition of the node
+    typedef Node<3> NodeType;
+
+    /// Pointer definition of CalculateDistanceToPathProcess
+    KRATOS_CLASS_POINTER_DEFINITION(CalculateDistanceToPathProcess);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /**
+     * @brief Default constructor.
+     * @param rModel The model containing the model parts
+     */
+    CalculateDistanceToPathProcess(
+        Model& rModel, 
+        Parameters ThisParameters
+        );
+
+    /// Destructor.
+    ~CalculateDistanceToPathProcess() override
+    {
+    }
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * @brief This method creates an pointer of the process
+     * @details We consider as input a Model and a set of Parameters for the sake of generality
+     * @warning Must be overrided in each process implementation
+     * @param rModel The model to be consider
+     * @param ThisParameters The configuration parameters
+     */
+    Process::Pointer Create(
+        Model& rModel,
+        Parameters ThisParameters
+        ) override
+    {
+        return Kratos::make_shared<CalculateDistanceToPathProcess>(rModel, ThisParameters);
+    }
+
+    /**
+     * @brief Execute method is used to execute the Process algorithms.
+     */
+    void Execute() override;
+
+    /**
+     * @brief Computes the minimal distance to a line with radius contribution
+     * @details Projects over a line and if the point projected is inside the line that distance is taken into consideration, otherwise the minimal between the two points in the line is considered
+     * @param rDistance The distance
+     * @param rSegment The line segment
+     * @param rPoint The point to compute distance
+     * @param Radius The radius
+     * @param Tolerance Tolerance to check it falls inside the line
+     * @return The Distance computation type
+     * @todo Add C++ test
+     */
+    static DistanceComputed FastMinimalDistanceOnLineWithRadius(
+        double& rDistance,
+        const Geometry<NodeType>& rSegment,
+        const Point& rPoint,
+        const double Radius,
+        const double Tolerance = 1.0e-9
+        );
```

Comment:
Why adding a `static` method to the process? This must be in a separated util IMO.